### ISSUE TITLE
Undefined symbols for architecture x86_64:

### ISSUE THREAD

**Issue by njafei** [CocoaPods/CocoaPods#6344]
# Report

## What did you do?
I want to submit  a new pod

ℹ Please replace this with what you did.  
pod lib lint

## What did you expect to happen?

success

## What happened instead?

failed 

## CocoaPods Environment
1.1.1

ℹ Please replace this with the output of `pod env`.

### Stack

```
   CocoaPods : 1.1.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.12.1 (16B2555)
       Xcode : 8.1 (8B62)
         Git : git version 2.9.3 (Apple Git-75)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 3dfbc2231315c345e26d698c2835f0d2859d95fc
               njafei - https://github.com/njafei/GTFramework.git @ 93e6efb5892ddeaea604009620d2d1e11cefa83b
               TFPaySDK - ssh://liufei@gerrit.tuniu.org:29418/fap-ios-TFPaySDK @ 59b321232222ef251e7b93ebd3bc2490cd60c9f8
               TNBase - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-ShellAPP @ 957d84a299cf766f2e8366254eeec1e624785bf2
               TNPhotographer - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-photographer @ f386a298452eec434e0c2d81de4b3adb29b8cb06
               TTNAirplane - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-airplane @ 7a90c3f0377c130dd42be77e54c3517cb4515603
               TTNBus - ssh://liufei@gerrit.tuniu.org:29418/app-ios-bus @ 3c433945d81d1b1a7b2571c0dec0d2fdbe6ad778
               TTNCar - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-lua @ b239ae761d5cf4ba3de6feb2eec5e759f78dfe30
               TTNChat - ssh://liufei@gerrit.tuniu.org:29418/app-ios-message @ 80acae729fa8f0c08c551429f5083b9e684746b2
               TTNCicerone - ssh://liufei@gerrit.tuniu.org:29418/app-ios-cicerone @ a766766c3be6e6a886998574a6e9d0612eabf049
               TTNComponent - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-component @ c4be5d4cbdbe0db51036435b649d585812998a32
               TTNCruiseShip - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-cruise @ 001c8ca90cb94978cc1b0c34ff2285653ee9a7d0
               TTNDiscovery - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-discovery @ cee75762b43780df0283a54cb31663309b23e002
               TTNDIY - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-diy @ 233418ee5456ca5e7186df80eeb1615566181505
               TTNHotel - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-hotel @ c3dcf2da181447999b20e527aa582de948fdbdbb
               TTNLogin - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-login @ 5adeaa3f6a4dbba06412c8ebed5ac25c675acf34
               TTNPackageTour - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-group @ e6ac2630d1aaf94ab56b9ca7541010ee65339396
               TTNSuper - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-superdiy @ 163f1881472431eb77a0bcd132283372208c48a2
               TTNTrain - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-train @ cb7551379b6b2ed19ee5ba36b83d80aa782e7c5d
               TTNUserCenter - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-usercenter @ 19edb9c4a63350775e69c1922879b96f71de9b79
               TTNVendor - ssh://liufei@gerrit.tuniu.org:29418/tuniu-ios-vendor @ 54a697de7f43df030fa04179e75579f511864cbb
               TTNWifi - ssh://liufei@gerrit.tuniu.org:29418/tuniuapp-iphone-wifi @ b680c91a28476ad4c4ff7a39b701a0b149f239af
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.1.2
cocoapods-try         : 1.1.0

## Project that demonstrates the issue

Undefined symbols for architecture x86_64:
  "_AudioServicesCreateSystemSoundID", referenced from:
      -[GTLogoWindow initWithFrame:delegate:] in GT
  "_AudioServicesDisposeSystemSoundID", referenced from:
      -[GTLogoWindow dealloc] in GT
  "_AudioServicesPlaySystemSound", referenced from:
      -[GTLogoWindow periodTimerNotify:] in GT
  "_CNCopyCurrentNetworkInfo", referenced from:
      -[GTMTAHelper fetchSSIDInfo] in GT
  "_CNCopySupportedInterfaces", referenced from:
      -[GTMTAHelper fetchSSIDInfo] in GT
  "_OBJC_CLASS_$_CTTelephonyNetworkInfo", referenced from:
      objc-class-ref in GT
  "_SCNetworkReachabilityCreateWithAddress", referenced from:
      -[GTNetModel currentNetStatus] in GT
      +[GTMTAReachability reachabilityWithAddress:] in GT
  "_SCNetworkReachabilityCreateWithName", referenced from:
      +[GTMTAReachability reachabilityWithHostname:] in GT
  "_SCNetworkReachabilityGetFlags", referenced from:
      -[GTNetModel currentNetStatus] in GT
      -[GTMTAReachability isReachable] in GT
      -[GTMTAReachability isReachableViaWWAN] in GT
      -[GTMTAReachability isReachableViaWiFi] in GT
      -[GTMTAReachability connectionRequired] in GT
      -[GTMTAReachability isConnectionOnDemand] in GT
      -[GTMTAReachability isInterventionRequired] in GT
      ...
  "_SCNetworkReachabilitySetCallback", referenced from:
      -[GTMTAReachability startNotifier] in GT
      -[GTMTAReachability stopNotifier] in GT
  "_SCNetworkReachabilitySetDispatchQueue", referenced from:
      -[GTMTAReachability startNotifier] in GT
      -[GTMTAReachability stopNotifier] in GT
  "_deflate", referenced from:
      +[GTMTAHelper GZip:Out:] in GT
  "_deflateEnd", referenced from:
      +[GTMTAHelper GZip:Out:] in GT
  "_deflateInit2_", referenced from:
      +[GTMTAHelper GZip:Out:] in GT
  "_inflate", referenced from:
      +[GTMTAHelper GUnzip:Out:] in GT
  "_inflateEnd", referenced from:
      +[GTMTAHelper GUnzip:Out:] in GT
  "_inflateInit2_", referenced from:
      +[GTMTAHelper GUnzip:Out:] in GT
  "_sqlite3_bind_int64", referenced from:
      ___35-[GTMTADataConfigHolder setConfig:]_block_invoke in GT
      ___26-[GTMTAStore storeConfig:]_block_invoke in GT
      ___28-[GTMTAStore storeEvent:CB:]_block_invoke in GT
  "_sqlite3_bind_text", referenced from:
      ___35-[GTMTADataConfigHolder setConfig:]_block_invoke in GT
      ___26-[GTMTAStore storeConfig:]_block_invoke in GT
      ___28-[GTMTAStore storeEvent:CB:]_block_invoke in GT
  "_sqlite3_clear_bindings", referenced from:
      ___35-[GTMTADataConfigHolder setConfig:]_block_invoke in GT
      ___26-[GTMTAStore storeConfig:]_block_invoke in GT
      ___28-[GTMTAStore storeEvent:CB:]_block_invoke in GT
  "_sqlite3_close", referenced from:
      -[GTMTADataConfigHolder closeDB] in GT
      -[GTMTAStore closeDB] in GT
  "_sqlite3_exec", referenced from:
      -[GTMTAStore updateDatabase:] in GT
      _execute_query in GT
      _execute_update in GT
  "_sqlite3_finalize", referenced from:
      ___35-[GTMTADataConfigHolder setConfig:]_block_invoke in GT
      ___26-[GTMTAStore storeConfig:]_block_invoke in GT
      ___28-[GTMTAStore storeEvent:CB:]_block_invoke in GT
  "_sqlite3_free", referenced from:
      -[GTMTAStore updateDatabase:] in GT
      _execute_query in GT
      _execute_update in GT
  "_sqlite3_open", referenced from:
      -[GTMTADataConfigHolder getDB] in GT
      -[GTMTAStore getDB] in GT
  "_sqlite3_prepare_v2", referenced from:
      ___35-[GTMTADataConfigHolder setConfig:]_block_invoke in GT
      ___26-[GTMTAStore storeConfig:]_block_invoke in GT
      ___28-[GTMTAStore storeEvent:CB:]_block_invoke in GT
  "_sqlite3_reset", referenced from:
      ___35-[GTMTADataConfigHolder setConfig:]_block_invoke in GT
      ___26-[GTMTAStore storeConfig:]_block_invoke in GT
      ___28-[GTMTAStore storeEvent:CB:]_block_invoke in GT
  "_sqlite3_step", referenced from:
      ___35-[GTMTADataConfigHolder setConfig:]_block_invoke in GT
      ___26-[GTMTAStore storeConfig:]_block_invoke in GT
      ___28-[GTMTAStore storeEvent:CB:]_block_invoke in GT
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **


The following build commands failed:
	Ld /Users/liufei/Library/Developer/Xcode/DerivedData/App-fjmtagbasblhskciodwhnwkkdzhu/Build/Intermediates/App.build/Release-iphonesimulator/App.build/Objects-normal/x86_64/App normal x86_64
(1 failure)
 -> GT (2.3.3)
    - WARN  | summary: The summary is not meaningful.
    - WARN  | description: The description is shorter than the summary.
    - WARN  | [iOS] license: Unable to find a license file
    - ERROR | [GT/GT] xcodebuild: Returned an unsuccessful exit code.
    - NOTE  | [GT/GT] xcodebuild:  clang: error: linker command failed with exit code 1 (use -v to see invocation)

[!] GT did not pass validation, due to 1 error and 3 warnings.
You can use the `--no-clean` option to inspect any issue.

/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/command/lib/lint.rb:79:in `block in run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/command/lib/lint.rb:50:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/command/lib/lint.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'


ℹ Please link to a project we can download that reproduces the issue.
You can delete this section if your issue is unrelated to build problems,
i.e. it's only an issue with CocoaPods the tool.

I upload framework and other resource , you can try it with:
https://github.com/njafei/GTFramework   on branch 2.3.3

**Comment by benasher44** [comment]
Hi @njafei! I looked at https://github.com/njafei/GTFramework, and there's no podspec there. Can you please provide the podspec that reproduces this issue? Thanks!

**Comment by njafei** [comment]
@benasher44  try https://github.com/njafei/GTFramework/tree/dev2.2.3

**Comment by dnkoutso** [comment]
@njafei at the very least you are missing the `SystemConfiguration.framework` and 'AudioToolbox.framework` added to your podspec.

You should add `s.frameworks = 'AudioToolbox', 'SystemConfiguration'`

This fixes some of the errors but not all, but you now get the point on how to get the remaining ones fixed.

**Comment by dnkoutso** [comment]
@njafei adding these lines to your podspec:

```    
s.frameworks = 'SystemConfiguration', 'AudioToolbox', 'CoreTelephony'
s.libraries = 'sqlite3', 'z'
```

Makes the build successful.

**Comment by njafei** [comment]
@dnkoutso  it works! how did you find out this? I didn't get it from log

**Comment by dnkoutso** [comment]
The symbols in the error message are not found during linking. Searching in which frameworks those symbols exist and then specifying them in the podspec makes it work.

Going to close this one now since there is no problem with CocoaPods here.

**Comment by njafei** [comment]
@dnkoutso thanks so much~

**Comment by MariamP** [comment]
@dnkoutso, @benasher44 I have the same problem with linting, but the case is little bit different. I have 2 SDKs and the second uses the classes defined in the first SDK. During the linting the undefined symbols issue is returned. This started happening after I've updated the Cocoapods version from 0.39.0 to 1.1.1. Is there any way to avoid that issue?

I've tried to compile the first SDK's source files in the second SDK, but is causes duplicate symbols issue.

Any help is very appreciated. Thanks in advance!)
